### PR TITLE
python3Packages.drivelib: init at 0.3.0, python3Packages.expiringdict: init at 1.2.1, git-annex-remote-googledrive: init at 1.3.0

### PR DESCRIPTION
**Pull request by gravndal**
###### Motivation for this change
#46554 doesn't look to have gone anywhere.

I've never added any packages to nixpkgs before, do tell me if I missed anything.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  propagatedBuildInputs = [ annexremote drivelib GitPython tenacity humanfriendly ];
+
+  # no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A git-annex special remote for Google Drive";
+    homepage = "https://pypi.org/project/git-annex-remote-googledrive/";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix:28]: Please add maintainers.

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  version = "1.2.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1031scjwd5mc2igayg4rbmvb89xdxf1fwkw2101kq3p0grvikp0c";
+  };
+
+  propagatedBuildInputs = [ annexremote drivelib GitPython tenacity humanfriendly ];
+
+  # no tests
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix:22]:
```suggestion
  # package does not contain tests
```

File: pkgs/development/python-modules/drivelib/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  propagatedBuildInputs = [
+    google_api_python_client
+    google-auth-oauthlib
+    google-auth-httplib2
+  ];
+
+  meta = with lib; {
+    description = "Easy access to the most common Google Drive API calls";
+    homepage = "https://pypi.org/project/drivelib/";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/drivelib/default.nix:27]: Please add yourself as a maintainer.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104919` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>gitAndTools.git-annex-remote-googledrive</li>
    <li>python37Packages.drivelib</li>
    <li>python38Packages.drivelib</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 104919` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>gitAndTools.git-annex-remote-googledrive</li>
    <li>python37Packages.drivelib</li>
    <li>python38Packages.drivelib</li>
  </ul>
</details>

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix (hunk 2 above)
Comment (gravndal) [reply pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix]: Changed it to the most common form:
```
$ rg 'doCheck = false' -C1 | rg '#.*$' | sed 's/.*\(#.*$\)/\1/' | sort | uniq -c | sort -h | tail
     11 # no tests implemented
     12 # tests not included with pypi release
     13 # no tests are present
     15 # error: invalid command 'test'
     18 # no tests included
     21 # No tests included
     25 # no tests
     30 # No tests
     46 # No tests in archive
     80 # has no tests
```

Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix]: You can do sort -u.

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+  propagatedBuildInputs = [ annexremote drivelib GitPython tenacity humanfriendly ];
+
+  # has no tests
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A git-annex special remote for Google Drive";
+    homepage = "https://pypi.org/project/git-annex-remote-googledrive/";
+    license = licenses.gpl3;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix:28]: Is it gpl3Only or gpl3Plus?

File: pkgs/development/python-modules/drivelib/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/drivelib/default.nix:27]: Is it gpl3Only or gpl3Plus?

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix (hunk 2 above)
Comment (gravndal) [reply pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix]: No, `sort -u` merely removes duplicates, it does not count the number of occurrences.

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix (hunk 4 above)
Comment (gravndal) [reply pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix]: I thought gpl3 implied that it wasn't an "or later"? Anyhow, changed it.

Comment (SuperSandro2000) [reply pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix]: gpl3 is ambiguous and in the way of being deprecated. Please help with that goal by not adding new occurrencs of deprecated licenses.

**Comment by gravndal** [comment]
Just noticed that the google-api-python-client package is broken:
```
Traceback (most recent call last):
  File "/nix/store/z9w9j2bhp4fimn1kcpw1ijyvzpgn5vib-git-annex-remote-googledrive-1.2.3/bin/.git-annex-remote-googledrive-wrapped", line 6, in <module>
    from git_annex_remote_googledrive.run import main
  File "/nix/store/z9w9j2bhp4fimn1kcpw1ijyvzpgn5vib-git-annex-remote-googledrive-1.2.3/lib/python3.8/site-packages/git_annex_remote_googledrive/run.py", line 22, in <module>
    from drivelib import __version__ as drivelib_version
  File "/nix/store/qpmhxwc88p0l59fy6ncpblkww879vh1s-drivelib-0.2.0/lib/python3.8/site-packages/drivelib/__init__.py", line 5, in <module>
    from .drive import *
  File "/nix/store/qpmhxwc88p0l59fy6ncpblkww879vh1s-drivelib-0.2.0/lib/python3.8/site-packages/drivelib/drive.py", line 15, in <module>
    from googleapiclient.discovery import build
  File "/nix/store/d7v837yx4mpi6arncnjjkgkgmppfzv8f-python3.8-google-api-python-client-1.12.8/lib/python3.8/site-packages/googleapiclient/discovery.py", line 68, in <module>
    from googleapiclient.http import build_http
  File "/nix/store/d7v837yx4mpi6arncnjjkgkgmppfzv8f-python3.8-google-api-python-client-1.12.8/lib/python3.8/site-packages/googleapiclient/http.py", line 67, in <module>
    from googleapiclient.model import JsonModel
  File "/nix/store/d7v837yx4mpi6arncnjjkgkgmppfzv8f-python3.8-google-api-python-client-1.12.8/lib/python3.8/site-packages/googleapiclient/model.py", line 30, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```
It's missing a dependency on setuptools.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 104919` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>gitAndTools.git-annex-remote-googledrive</li>
<li>python37Packages.drivelib</li>
<li>python38Packages.drivelib</li>
<li>python39Packages.drivelib</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.drivelib:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python38Packages.drivelib:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>
python39Packages.drivelib:
<details>

Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
You have the following options to fix this:
  - Tell pytest(CheckHook) where to find the tests included in the package
  - Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
  - If the Packages does not contain any tests add \'doCheck = false;'` and a `pythonImportsCheck`.

</details>

**Comment by SuperSandro2000** [comment]
> Just noticed that the google-api-python-client package is broken:

I just merged the PR that fixes it.

**Comment by gravndal** [comment]
> Zero tests run by pytest got detected: 'Ran 0 tests in 0.000s'
> You have the following options to fix this:
> 
>     * Tell pytest(CheckHook) where to find the tests included in the package
> 
>     * Check if the GitHub Repo contains tests but they are not shipped with Pypi. If so please switch to `fetchFromGitHub`.
> 
>     * If the Packages does not contain any tests add 'doCheck = false;'`and a`pythonImportsCheck`.

Ah, I missed that, sorry. Had thought tests ran as the build never complained unlike with git-annex-remote-googledrive, and I see now that I confused the build output of tenacity as a part of drivelib.

Switching over to `fetchFromGithub` tests now run, but they (pretty obviously) need a google auth token to pass, so I'll disable them.

**Comment by gravndal** [comment]
Sorry about the delay here.

File: pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "118w0fyy6pck8hyj925ym6ak0xxqhkaq2vharnpl9b97nab4mqg8";
+  };
+
+  propagatedBuildInputs = [ annexremote drivelib GitPython tenacity humanfriendly ];
+
+  # while git-annex does come with a testremote command that *could* be used,
+  # testing this special remote obviously depends on authenticating with google
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/git-and-tools/git-annex-remote-googledrive/default.nix:24]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "git_annex_remote_googledrive" ];
```

File: pkgs/development/python-modules/drivelib/default.nix [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    google-api-python-client
+    google-auth-oauthlib
+    google-auth-httplib2
+    expiringdict
+  ];
+
+  # tests depend on a google auth token
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/drivelib/default.nix:27]:
```suggestion
  doCheck = false;
  pythonImportsCheck = [ "drivelib" ];
```

File: pkgs/development/python-modules/expiringdict/default.nix [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    dill
+    coverage
+    coveralls
+    mock
+    nose
+  ];
+
+  checkPhase = ''
+    nosetests -v --with-coverage --cover-package=expiringdict
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/expiringdict/default.nix:34]:
```suggestion
  '';
  
  pythonImportsCheck = [ "expiringdict" ];
```

**Pull request merged**